benchmarks/RequestResponseLatency.cpp

```diff
@@ -0,0 +1,228 @@
+public:
+    ~BM_Subscriber()
+    {
+        LOG(INFO) << "BM_Subscriber destroy " << this;
+    }
+
+    BM_Subscriber()
+        : initialRequest_(8),
+        thresholdForRequest_(initialRequest_ * 0.75)
+    {
+        LOG(INFO) << "BM_Subscriber " << this << " created with => "
+            << "  Initial Request: " << initialRequest_
+            << "  Threshold for re-request: " << thresholdForRequest_;
+    }
+
+    void onSubscribe(std::shared_ptr<reactivesocket::Subscription> subscription) noexcept override
+    {
+        LOG(INFO) << "BM_Subscriber " << this << " onSubscribe";
+        subscription_ = std::move(subscription);
+        requested_ = initialRequest_;
+        subscription_->request(initialRequest_);
+    }
+
+    void onNext(reactivesocket::Payload element) noexcept override
+    {
+        LOG(INFO) << "BM_Subscriber " << this
+            << " onNext as string: " << element.moveDataToString();
+
+        if (--requested_ == thresholdForRequest_) {
+            int toRequest = (initialRequest_ - thresholdForRequest_);
+            LOG(INFO) << "BM_Subscriber " << this << " requesting " << toRequest
+                << " more items";
+            requested_ += toRequest;
+            subscription_->request(toRequest);
+        }
+    }
+
+    void onComplete() noexcept override
+    {
+        LOG(INFO) << "BM_Subscriber " << this << " onComplete";
+        terminated_ = true;
+        completed_ = true;
+        terminalEventCV_.notify_all();
+    }
+
+    void onError(folly::exception_wrapper ex) noexcept override
+    {
+        LOG(INFO) << "BM_Subscriber " << this << " onError " << ex.what();
+        terminated_ = true;
+        terminalEventCV_.notify_all();
+    }
+
+    void awaitTerminalEvent()
+    {
+        LOG(INFO) << "BM_Subscriber " << this << " block thread";
+        // now block this thread
+        std::unique_lock<std::mutex> lk(m_);
+        // if shutdown gets implemented this would then be released by it
+        terminalEventCV_.wait(lk, [this] { return terminated_; });
+        LOG(INFO) << "BM_Subscriber " << this << " unblocked";
+    }
+
+    bool completed()
+    {
+        return completed_;
+    }
+
+private:
+    int initialRequest_;
+    int thresholdForRequest_;
+    int requested_;
+    std::shared_ptr<reactivesocket::Subscription> subscription_;
+    bool terminated_{false};
+    std::mutex m_;
+    std::condition_variable terminalEventCV_;
+    std::atomic_bool completed_{false};
+};
+
+class BM_RsFixture : public benchmark::Fixture
+{
+public:
+    BM_RsFixture() :
+        host_(FLAGS_host),
+        port_(FLAGS_port),
+        serverRs_(RSocket::createServer(TcpConnectionAcceptor::create(port_))),
+        handler_(std::make_shared<BM_RequestHandler>())
+    {
+        FLAGS_v = 0;
+        FLAGS_minloglevel = 6;
+        serverRs_->start([this](auto r) { return handler_; });
+    }
+
+    virtual ~BM_RsFixture()
+    {
+    }
+
+    virtual void SetUp(benchmark::State &state)
```

Comment:
Better to make them go away as I believe they are null... they are only left for those new to GB and perhaps wanting to add state. They are out of the measurement path, so they don't hurt. If stylistically, `override` is preferred, go for it.